src/widget/wcolorpicker.cpp

```diff
@@ -30,96 +54,155 @@ WColorPicker::WColorPicker(ColorOption colorOption, QWidget* parent)
     // better than having buttons without any colors (which would make the
     // color picker unusable).
     m_pStyle = QStyleFactory::create(QString("fusion"));
+    setLayout(pLayout);
+    addColorButtons();
+}
+
+void WColorPicker::removeColorButtons() {
+    QGridLayout* pLayout = static_cast<QGridLayout*>(layout());
+    VERIFY_OR_DEBUG_ASSERT(pLayout) {
+        qWarning() << "Color Picker has no layout!";
```

Comment:
would this be a problem with the skin?  Maybe a hint:  "skin design error: color picker has no layout"

Reply:
This should not happen unless somebody unsets the layout from C++ code.

Reply:
is that something that can be tested from a unit test rather than a runtime test?

Reply:
No. I can just remove that test altogether if you want.

Reply:
no that's ok.  in areas where we can't have test coverage, it's better than nothing to have a developer runtime check.  We don't want this to be the primary way we do regression testing, but it's all we've got